### DISCUSSION TITLE
Lesson 10: "ValueError: Unable to expand environment..."

### DISCUSSION THREAD

**Discussion by yellowcheese773** [smartcontractkit/full-blockchain-solidity-course-py#114] [Q&A] [answered]
I did copy exactly every step from Patrick, unfortunately, I get this error, when I do:

 `brownie run scripts/get_weth.py --network kovan`

<img width="1241" alt="Bildschirmfoto 2021-09-29 um 13 59 21" src="https://user-images.githubusercontent.com/80122924/135264429-21b5081b-2db8-4854-a105-68a493855300.png">

I don't know how to fix this. 

Here is my brownie-config: 

<img width="1241" alt="Bildschirmfoto 2021-09-29 um 13 57 39" src="https://user-images.githubusercontent.com/80122924/135264659-65b58351-b5a8-4106-816d-482908a12cff.png">

I would be really thankful, if someone can help me.

**Comment by toddabraham** [comment]
Do you mind sharing the contents of your helpful_scripts.py file?  It maybe that you need to add network, and config to your `from brownie import...` line.  It should look this this `from brownie import accounts, network, config`.

**Reply by yellowcheese773** [reply]
Of course, here is my helpful_scripts.py file: 

<img width="1241" alt="Bildschirmfoto 2021-10-01 um 14 54 58" src="https://user-images.githubusercontent.com/80122924/135623122-98f58712-35f4-4c85-93a0-a24d51827824.png">

I noticed that `brownie import`, "brownie" is not colored, or is flagged as a problem: 

<img width="1241" alt="Bildschirmfoto 2021-10-01 um 14 57 27" src="https://user-images.githubusercontent.com/80122924/135623561-5dc74cff-f925-4d69-bd2f-4915eca111df.png">

Could that be a serious problem?

**Reply by toddabraham** [reply]
You have forgotten an `e` in brownie.  That may be the issue.

![image](https://user-images.githubusercontent.com/26919212/135698085-3551f837-152f-43ad-acbb-ab643b3cc810.png)

**Comment by toddabraham** [comment]
Ah, but I noticed in your previous screen shot, you have `brownie` spelled correctly.

![image](https://user-images.githubusercontent.com/26919212/135698147-0fee0475-80d9-4329-ad4f-7b8cacceace9.png)

**Reply by toddabraham** [reply]
Maybe check to make sure you have the right INFURA Project ID in your .env file.

![image](https://user-images.githubusercontent.com/26919212/135698273-c9ab6453-ec37-4c3b-a85f-f492289d89f1.png)

**Reply by yellowcheese773** [reply]
I checked my Infura project Id, i even created a new project Id and put that in the env and tried it again. Unfortunately it did not work either. I guess i have to go through all of it again and find the problem. Anyway, thanks for your help mate.

**Comment by PatrickAlphaC** [comment, marked as answer]
It looks like your `$WEB3_INFURA_PROJECT_ID` is undefined. Do you have that in your `.env` file like in this repo? 

You'll also need the line `dotenv: .env` in your config. 

https://github.com/PatrickAlphaC/aave_brownie_py/blob/main/.env

You'll need to uncomment the lines (remove the #'s)

**Reply by yellowcheese773** [reply]
Ive looked again at everything, at the extensions and settings... 

I had uninstalled Solidity for whatever reason, maybe that is causing those problems. Thanks for your help Patrick, I am gonna look into that .env file again.

**Reply by PatrickAlphaC** [reply]
Ok. Could you let us know if/when you resolve it? It should just be that your environment variables are not being set properly. This is a pretty common error. What's your `.env` file look like?

**Reply by Daruma1800** [reply]
I have been having an issue were my .env file isn't even recognized, the export string isn't highlighted and have been stuck on this for awhile now, I have deleted the file and redid it a few times now. Have you guys run into this problem before? 

![image](https://user-images.githubusercontent.com/94502087/150653066-fe531ac1-8d46-42a4-8f01-e74332e781aa.png)

**Reply by Daruma1800** [reply]
@PatrickAlphaC @cromewar forgot to tag you both here, also not to worry this is a dumby metamask account for this course.

**Reply by cromewar** [reply]
Make sure to `source .env` on your terminal to get the env variables, also you can use the `export` command directly into the terminal with the private keys, give it a try.

**Reply by PatrickAlphaC** [reply]
Or add `dotenv: .env` to your `brownie-config.yaml`

**Comment by Daruma1800** [comment]
Thanks guys, I just set up a new brownie init and it’s been working but
still no highlight for the .env file.

**Reply by cromewar** [reply]
As Patrick said, did you add dotenv to brownie-config.yaml?
